internal/metrics/logfilemetricexporter/factory.go

```diff
@@ -0,0 +1,120 @@
+	"github.com/openshift/cluster-logging-operator/internal/collector"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	coreFactory "github.com/openshift/cluster-logging-operator/internal/factory"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+)
+
+const (
+	clusterLoggingPriorityClassName = "system-node-critical"
+	ExporterPort                    = int32(2112)
+	ExporterPortName                = "exporter-port"
+	exporterMetricsVolumeName       = "lfme-metrics"
+	ExporterMetricsSecretName       = "lfme-secret"
+	logContainers                   = "varlogcontainers"
+	logContainersValue              = "/var/log/containers"
+	logPods                         = "varlogpods"
+	logPodsValue                    = "/var/log/pods"
+	metricsVolumePath               = "/etc/logfilemetricexporter/metrics"
+)
+
+var (
+	defaultTolerations = []v1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		},
+		{
+			Key:      "node.kubernetes.io/disk-pressure",
+			Operator: v1.TolerationOpExists,
+			Effect:   v1.TaintEffectNoSchedule,
+		},
+	}
+)
+
+// ResourceRequirements returns the resource requirements for a given metric-exporter implementation
+// or it's default if none are specified
+func ResourceRequirements(exporter loggingv1a1.LogFileMetricExporter) v1.ResourceRequirements {
+	if exporter.Spec.Resources == nil {
+		return v1.ResourceRequirements{}
+	}
+	return *exporter.Spec.Resources
+}
+
+func NodeSelector(exporter loggingv1a1.LogFileMetricExporter) map[string]string {
+	return exporter.Spec.NodeSelector
+}
+func Tolerations(exporter loggingv1a1.LogFileMetricExporter) []v1.Toleration {
+	if exporter.Spec.Tolerations == nil {
+		return defaultTolerations
+	}
+	return append(defaultTolerations, exporter.Spec.Tolerations...)
+}
+
+func NewDaemonSet(exporter loggingv1a1.LogFileMetricExporter, namespace, name string, collectionType loggingv1.LogCollectionType, tlsProfileSpec configv1.TLSProfileSpec) *apps.DaemonSet {
+	podSpec := NewPodSpec(exporter, tlsProfileSpec)
+	ds := coreFactory.NewDaemonSet(name, namespace, constants.LogfilesmetricexporterName, constants.LogfilesmetricexporterName, string(collectionType), *podSpec)
+	return ds
+}
+
+func NewPodSpec(exporter loggingv1a1.LogFileMetricExporter, tlsProfileSpec configv1.TLSProfileSpec) *v1.PodSpec {
+
+	podSpec := &v1.PodSpec{
+		NodeSelector:                  utils.EnsureLinuxNodeSelector(NodeSelector(exporter)),
+		PriorityClassName:             clusterLoggingPriorityClassName,
+		ServiceAccountName:            constants.CollectorServiceAccountName,
+		TerminationGracePeriodSeconds: utils.GetInt64(10),
+		Tolerations:                   Tolerations(exporter),
+		Volumes: []v1.Volume{
+			{Name: logContainers, VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: logContainersValue}}},
```

Comment:
@alanconway @vimalk78 are we still reading from /var/log/containers?  Can we remove this mount point?

Reply:
following up here I believe via slack we confirmed we are still using this path